internal/access/credential.go

```diff
@@ -86,10 +86,16 @@ func ResetCredential(rCtx RequestContext, user *models.Identity, newPassword str
 		return "", fmt.Errorf("get credential: %w", err)
 	}
 
+	hash, err := bcrypt.GenerateFromPassword([]byte(newPassword), bcrypt.DefaultCost)
```

Comment:
This hashing feels like it might make more sense in the `server` or `data` packages rather than the `access` check, although I see it was here previously too

Reply:
`access` seems to be the right place. It's where a lot of decision regarding password generation and policy are made. If this hashing is done too early (in `server),`, there's not a lot that can be done wrt password policy. If it's done too late (in `data)`, many of these decision need to be delayed or exceptions has to be made.